easybuild/easyblocks/t/tensorflow.py

```diff
@@ -701,13 +701,22 @@ def configure_step(self):
                 })
             else:
                 raise EasyBuildError("TensorFlow has a strict dependency on cuDNN if CUDA is enabled")
+
             if nccl_root:
                 nccl_version = get_software_version('NCCL')
                 # Ignore the PKG_REVISION identifier if it exists (i.e., report 2.4.6 for 2.4.6-1 or 2.4.6-2)
                 nccl_version = nccl_version.split('-')[0]
                 config_env_vars.update({
                     'NCCL_INSTALL_PATH': nccl_root,
                 })
+
+                # add path to libnccl.so.2 directory provided by NCCL when both sysroot
+                # and RPATH are used (such as in EESSI)
+                if build_option('sysroot') and self.toolchain.use_rpath:
```

Comment:
Is `build_option('sysroot')` a necessary condition here? The core issue is use of rpath and lack of `LD_LIBRARY_PATH`

Reply:
This is a general issue though, if you have that configuration (rpath and no `LD_LIBRARY_PATH`) then you will run into this problem. The solution is specific to NCCL, and that is fine. I wouldn't introduce something arbitrary that would affect reproducibility.

You could use something similar to https://github.com/easybuilders/easybuild-easyblocks/blob/2ab3cbc3a542f1b5d31052ad6f28a39ffcc3fec0/easybuild/easyblocks/p/python.py#L199-L200

Reply:
Indeed, if this is not specific to using an alternate sysroot, then remove that part of the condition, there's no need to artificially restrict this to the EESSI context...

Reply:
That's right, it seems not specific to using an alternate sysroot.

Changed the conditional expression (see https://github.com/easybuilders/easybuild-easyblocks/pull/3497/commits/edc9bfe6456d573d6ab0a95f37ea87962965b5a1) and tested this. After the change, `libnccl.so.2` is found.

Note, there is another issue building TensorFlow with EESSI, but I may know how to fix that. It likely requires changing the shebang so scripts use `env` from the compat layer. Changing the scripts will need some addition work to circumvent some sanity check run by Bazel (see https://stackoverflow.com/questions/47775668/bazel-how-to-skip-corrupt-installation-on-centos6).